### PR TITLE
Remove browser bundles and TypeScript tests

### PR DESCRIPTION
**Pull request by pepone**
This PR removes support for browser bundles and IIFE (Immediately Invoked Function Expression) modules. Additionally, it includes cleanup of files that were previously generated from local Slice.

Ice remains a CommonJS module. The next step will be to convert Ice to an ES6 module and subsequently remove support for CommonJS modules.

### CODE REVIEW

File: js/src/Ice/LocalException.d.ts [hunk 1]
```diff
@@ -18,20 +18,6 @@ export namespace Ice
         reason:string;
     }
```
Comment (pepone) [inline js/src/Ice/LocalException.d.ts:20]: Removed several unused local exceptions in the JavaScript codebase.

File: js/test/Glacier2/router/Client.js [hunk 2]
```diff
@@ -2,335 +2,327 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
-(function(module, require, exports)
```
Comment (pepone) [inline js/test/Glacier2/router/Client.js:5]: For this tests an others, I just remove the wrapping IIFE, they are now Common JS modules and an follow up PR would convert them to ES6 modules.

File: js/test/typescript/Glacier2/router/Client.ts
Comment (pepone) [inline js/test/typescript/Glacier2/router/Client.ts:1]: Removed the TypeScript tests. Maintaining duplicated tests for TypeScript offers little value, as both sets of tests target the same Ice for JavaScript library and generated code.

**Comment by bernardnormier** [comment]
I don't think we should remove all the TypeScript tests, because we need to test somewhere we provide or generate expected/usable TypeScript definitions.

Removing the TypeScript tests would definitely reduce our test coverage.

Maybe we should remove the (duplicated) JavaScript tests. The TS test for feature F presumably tests both the TS API for F, the runtime behavior for F (in JS) and indirectly the JS API.

**Pull request closed without merging**